clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -208,12 +208,9 @@ private void maybeRemoveAndDeallocateBatch(ProducerBatch batch) {
 
     private void addToInflightBatches(List<ProducerBatch> batches) {
         for (ProducerBatch batch : batches) {
-            List<ProducerBatch> inflightBatchList = inFlightBatches.get(batch.topicPartition);
-            if (inflightBatchList == null) {
-                inflightBatchList = new ArrayList<>();
-                inFlightBatches.put(batch.topicPartition, inflightBatchList);
-            }
-            inflightBatchList.add(batch);
+            Objects.requireNonNull(inFlightBatches.computeIfAbsent(batch.topicPartition,
```

Comment:
Why do we need `requireNonNull` here?

Reply:
Sorry I didn't think over it. `requireNonNull` is redundant.